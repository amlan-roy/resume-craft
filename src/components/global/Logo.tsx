import Image from "next/image";
import React from "react";
import Link from "next/link";
import { cn } from "@/lib/utils";

type LogoProps = {
  linkify?: boolean;
  hideText?: boolean;
  className?: string;
  imageClassName?: string;
  textClassName?: string;
};

const LogoContainer: React.FC<{
  linkify?: boolean;
  children?: React.ReactNode;
  className?: string;
}> = ({ linkify, children, className }) => {
  return linkify ? (
    <Link
      data-testid="global__logo-linkified"
      href={"/"}
      className={cn(["flex justify-center items-center", className])}
    >
      {children}
    </Link>
  ) : (
    <div
      data-testid="global__logo"
      className={cn(["flex justify-center items-center", className])}
    >
      {children}
    </div>
  );
};

const Logo: React.FC<LogoProps> = ({
  linkify,
  hideText,
  className,
  imageClassName,
  textClassName,
}) => {
  return (
    <LogoContainer linkify={linkify} className={className}>
      <Image
        src={"/icon.png"}
        alt={"Resume Craft Icon"}
        width={36}
        height={36}
        className={cn(["w-9 h-9 sm:w-full sm:h-full", imageClassName])}
      />
      {!hideText && (
        <div
          className={cn([
            "flex flex-col text-brand-neutral-12 mx-3 justify-center items-center",
            textClassName,
          ])}
        >
          <p className="text-base sm:text-xl font-extrabold leading-none sm:leading-none">
            RESUME
          </p>
          <p className="text-base sm:text-xl leading-none sm:leading-none ">
            <span className="font-light"> - </span>craft
            <span className="font-light"> - </span>
          </p>
        </div>
      )}
    </LogoContainer>
  );
};
export default Logo;
