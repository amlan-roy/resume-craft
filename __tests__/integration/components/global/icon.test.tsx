import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Logo from "@/components/global/Logo";

describe("integration:components/global/icon", () => {
  it("snapshot", () => {
    const tree = render(<Logo />);
    expect(tree.container).toMatchSnapshot();
  });

  it("Renders the logo with the image and text by default", () => {
    render(<Logo />);

    const logoImage = screen.getByAltText("Resume Craft Icon");
    const resumeText = screen.getByText("RESUME");
    const craftText = screen.getByText("craft");

    expect(logoImage).toBeInTheDocument();
    expect(resumeText).toBeInTheDocument();
    expect(craftText).toBeInTheDocument();
  });

  it("Renders the logo as a link when linkify is true", () => {
    render(<Logo linkify />);

    const linkifiedLogo = screen.getByTestId("global__logo-linkified");

    expect(linkifiedLogo).toBeInTheDocument();
    expect(linkifiedLogo).toHaveAttribute("href", "/");
  });

  it("Does not render the text when hideText is true", () => {
    render(<Logo hideText />);

    const logoImage = screen.getByAltText("Resume Craft Icon");
    const resumeText = screen.queryByText("RESUME");

    expect(logoImage).toBeInTheDocument();
    expect(resumeText).not.toBeInTheDocument();
  });
});
