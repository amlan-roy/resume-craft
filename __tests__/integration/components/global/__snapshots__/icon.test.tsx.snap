// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/icon snapshot 1`] = `
<div>
  <div
    class="flex justify-center items-center"
    data-testid="global__logo"
  >
    <img
      alt="Resume Craft Icon"
      class="w-9 h-9 sm:w-full sm:h-full"
      data-nimg="1"
      decoding="async"
      height="36"
      loading="lazy"
      src="/_next/image?url=%2Ficon.png&w=96&q=75"
      srcset="/_next/image?url=%2Ficon.png&w=48&q=75 1x, /_next/image?url=%2Ficon.png&w=96&q=75 2x"
      style="color: transparent;"
      width="36"
    />
    <div
      class="flex flex-col text-brand-neutral-12 mx-3 justify-center items-center"
    >
      <p
        class="text-base sm:text-xl font-extrabold leading-none sm:leading-none"
      >
        RESUME
      </p>
      <p
        class="text-base sm:text-xl leading-none sm:leading-none "
      >
        <span
          class="font-light"
        >
           - 
        </span>
        craft
        <span
          class="font-light"
        >
           - 
        </span>
      </p>
    </div>
  </div>
</div>
`;
